backend/lib/endpoints/posts.js

```diff
@@ -127,19 +216,102 @@ async function routes(app) {
 
   app.patch(
     "/:postId",
-    { preValidation: [app.authenticate], schema: updatePostSchema },
+    {
+      preValidation: [app.authenticate],
+      schema: updatePostSchema,
+    },
     async (req, reply) => {
-      // todo: make sure user can only update their own post
+      // TODO: get userId from jwt
+      // userId = ?
+      // user = User.findById(userId);
+      var user = await User.findById("5ea6900c0e0419d4cb123611");
+      if (user === null) return new httpErrors.Unauthorized();
+
       const post = await Post.findById(req.params.postId);
-      if (post === null) {
-        return reply.send(new httpErrors.NotFound());
-      }
-      Object.keys(req.body).forEach((key) => {
-        if (post[key] && post[key] !== req.body[key]) {
-          post[key] = req.body[key];
+      if (!post) return new httpErrors.NotFound();
+      else if (post.author.authorId != user.id)
+        return new httpErrors.Forbidden();
+      var body = req.body;
+
+      // ExpireAt needs to calculate the date
+      if ("expireAt" in body) {
+        switch (body.expireAt) {
+          case "day":
+            body.expireAt = (1).days().fromNow();
+            break;
+          case "week":
+            body.expireAt = (7).days().fromNow();
+            break;
+          case "month":
+            body.expireAt = (1).months().fromNow();
+            break;
         }
-      });
-      return post.save();
+      }
+
+      return Object.assign(post, body).save();
+    },
+  );
+
+  app.put(
+    "/:postId/likes/:userId",
+    {
+      preValidation: [app.authenticate],
+      schema: likeUnlikePostSchema,
+    },
+    async (req) => {
+      // TODO: get userId from jwt
+      // userId = ?
+      // user = User.findById(userId);
+      var user = await User.findById("5ea6900c0e0419d4cb123611");
+      if (user === null) return new httpErrors.Unauthorized();
+
+      const { postId, userId } = req.params;
+      if (postId === null || userId === null)
+        return new httpErrors.BadRequest();
```

Comment:
so I think these things are handled by the auth and schema validation now (and on my recent PR now on master I just checked with req.userId here)
If these aren't currently handled by `null` values in the request we might need to adjust the schema's, I 'm not sure if the default is `nullable`